raster-foundry/raster-foundry
# Add Layer Search Query Parameter + Update Frontend Filters

## Overview

This PR adds support for limiting scene search based on a layer AOI via a new query parameter. Additionally, this updates the frontend to use the query parameter, the filter styling, and to enable the "unprocessed" filter.

### Checklist


### Demo

![demo](https://user-images.githubusercontent.com/898060/53883700-b351f980-3fe7-11e9-9b09-8e29ed2681b8.gif)

### Notes

The new unprocessed filter ended up kind of tricky to add. The difficulty is that when the checkbox for "Show Unprocessed" is _not_ checked we need to add a query parameter (`ingested=true`) to the API call and when it _is_ checked we remove that query parameter.


## Testing Instructions

 * Open up a project in the new UI
 * Go to add imagery via browsing (you will be prompted to define an AOI if you have yet to do so)
 * Realize that the bounding box no long affects scenes returned because the scene results are being filtered by the layer AOI
 * Check that filters still work and that by checking the new "Show Unprocessed" filter more scenes are returned

Closes #4672 
Closes #4670
